### ISSUE TITLE
RubyGems fails

### ISSUE THREAD

**Issue by loranger** [topgrade-rs/topgrade#247]
## What did you expect to happen?

I would expect topgrade to upgrade everything flawlessyl, as usual

## What actually happened?

Since few days, I get an error while upgrading gem

<details>
<summary>output</summary>

```
―― 20:26:54 - RubyGems ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――
ERROR:  While executing gem ... (Gem::UnknownCommandError)
    Unknown command update --system
	/Users/loranger/.rubies/ruby-3.1.0/lib/ruby/site_ruby/3.1.0/rubygems/command_manager.rb:198:in `find_command'
	/Users/loranger/.rubies/ruby-3.1.0/lib/ruby/site_ruby/3.1.0/rubygems/command_manager.rb:183:in `process_args'
	/Users/loranger/.rubies/ruby-3.1.0/lib/ruby/site_ruby/3.1.0/rubygems/command_manager.rb:149:in `run'
	/Users/loranger/.rubies/ruby-3.1.0/lib/ruby/site_ruby/3.1.0/rubygems/gem_runner.rb:51:in `run'
	/Users/loranger/.rubies/ruby-3.1.0/bin/gem:13:in `<main>'
rubygems failed:
   0: Command failed: `/Users/loranger/.rubies/ruby-3.1.0/bin/gem 'update --system' --user-install`
   1: `/Users/loranger/.rubies/ruby-3.1.0/bin/gem` failed: exit status: 1

Location:
   src/steps/generic.rs:100

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
Retry? (y)es/(N)o/(s)hell/(q)uit
```

</details>

## Additional Details
- OS: macOS Ventura 13.0.1
- Topgrade installed via brew
- Topgrade 10.2.1

<details>
<summary>topgrade --dry-run</summary>

```
―― 20:33:23 - Brew ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Dry running: /opt/homebrew/bin/brew update
Dry running: /opt/homebrew/bin/brew upgrade --ignore-pinned --formula

―― 20:33:23 - Brew - Cask ――――――――――――――――――――――――――――――――――――――――――――――――――――――
Dry running: /opt/homebrew/bin/brew upgrade --cask

―― 20:33:24 - Git repositories ―――――――――――――――――――――――――――――――――――――――――――――――――
Would pull /Users/loranger/.zprezto

―― 20:33:24 - pip3 ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Dry running: /opt/homebrew/bin/python3 -m pip install --upgrade --user pip

―― 20:33:24 - Node Package Manager ―――――――――――――――――――――――――――――――――――――――――――――
Dry running: /opt/homebrew/bin/npm update '--location=global'

―― 20:33:25 - Composer ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Dry running: /opt/homebrew/bin/composer global update

―― 20:33:25 - Gems ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Dry running: /Users/loranger/.rubies/ruby-3.1.0/bin/gem update --user-install

―― 20:33:25 - RubyGems ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Dry running: /Users/loranger/.rubies/ruby-3.1.0/bin/gem 'update --system' --user-install

―― 20:33:25 - Local Nextcloud Apps (docker) ――――――――――――――――――――――――――――――――――――
Dry running: /bin/zsh -c 'ssh rock.local '\''docker-compose -f /home/loranger/nextcloud/docker-compose.yml run --rm occ app:update --all'\'''

―― 20:33:25 - Nextcloud (ansible) ――――――――――――――――――――――――――――――――――――――――――――――
Dry running: /bin/zsh -c 'ansible nextcloud -m shell -a '\''php /var/www/nextcloud/updater/updater.phar --no-interaction'\'' --become --become-user nextcloud'

―― 20:33:25 - Nextcloud Apps (ansible) ―――――――――――――――――――――――――――――――――――――――――
Dry running: /bin/zsh -c 'ansible nextcloud -m shell -a '\''php /var/www/nextcloud/occ app:update --all'\'' --become --become-user nextcloud'

―― 20:33:25 - macOS App Store ――――――――――――――――――――――――――――――――――――――――――――――――――
Dry running: /opt/homebrew/bin/mas upgrade

―― 20:33:25 - macOS system update ――――――――――――――――――――――――――――――――――――――――――――――
Finding available software
No new software available.

―― 20:33:28 - Summary ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Brew (ARM): OK
Brew Cask (ARM): OK
Git repositories: OK
pip3: OK
npm: OK
composer: OK
gem: OK
rubygems: OK
Local Nextcloud Apps (docker): OK
Nextcloud (ansible): OK
Nextcloud Apps (ansible): OK
App Store: OK
System upgrade: OK
```
</details>

<details>
<summary>topgrade -v</summary>

```
(...)
DEBUG Step "krew"
DEBUG Step "gem"
DEBUG Detected "/Users/loranger/.rubies/ruby-3.1.0/bin/gem" as "gem"
DEBUG Path "/Users/loranger/.gem" exists

―― 20:35:52 - Gems ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
DEBUG Detected rbenv. Avoiding --user-install
DEBUG Executing command `/Users/loranger/.rubies/ruby-3.1.0/bin/gem update --user-install`
Updating installed gems
Nothing to update
DEBUG Step "rubygems"
DEBUG Detected "/Users/loranger/.rubies/ruby-3.1.0/bin/gem" as "gem"
DEBUG Path "/Users/loranger/.gem" exists

―― 20:35:54 - RubyGems ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――
DEBUG Detected rbenv. Avoiding --user-install
DEBUG Executing command `/Users/loranger/.rubies/ruby-3.1.0/bin/gem 'update --system' --user-install`
ERROR:  While executing gem ... (Gem::UnknownCommandError)
    Unknown command update --system
	/Users/loranger/.rubies/ruby-3.1.0/lib/ruby/site_ruby/3.1.0/rubygems/command_manager.rb:198:in `find_command'
	/Users/loranger/.rubies/ruby-3.1.0/lib/ruby/site_ruby/3.1.0/rubygems/command_manager.rb:183:in `process_args'
	/Users/loranger/.rubies/ruby-3.1.0/lib/ruby/site_ruby/3.1.0/rubygems/command_manager.rb:149:in `run'
	/Users/loranger/.rubies/ruby-3.1.0/lib/ruby/site_ruby/3.1.0/rubygems/gem_runner.rb:51:in `run'
	/Users/loranger/.rubies/ruby-3.1.0/bin/gem:13:in `<main>'
DEBUG Command failed: Err(
   0: Command failed: `/Users/loranger/.rubies/ruby-3.1.0/bin/gem 'update --system' --user-install`
   1: `/Users/loranger/.rubies/ruby-3.1.0/bin/gem` failed: exit status: 1

Location:
   src/steps/generic.rs:100

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.)
DEBUG Step "rubygems" failed:
   0: Command failed: `/Users/loranger/.rubies/ruby-3.1.0/bin/gem 'update --system' --user-install`
   1: `/Users/loranger/.rubies/ruby-3.1.0/bin/gem` failed: exit status: 1

Location:
   src/steps/generic.rs:100

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
rubygems failed:
   0: Command failed: `/Users/loranger/.rubies/ruby-3.1.0/bin/gem 'update --system' --user-install`
   1: `/Users/loranger/.rubies/ruby-3.1.0/bin/gem` failed: exit status: 1

Location:
   src/steps/generic.rs:100

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.DEBUG Desktop notification: rubygems failed

Retry? (y)es/(N)o/(s)hell/(q)uit
```

</details>

<details>
<summary>RUST_BACKTRACE=1 topgrade --only ruby_gems</summary>

```
―― 20:41:17 - RubyGems ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――
ERROR:  While executing gem ... (Gem::UnknownCommandError)
    Unknown command update --system
	/Users/loranger/.rubies/ruby-3.1.0/lib/ruby/site_ruby/3.1.0/rubygems/command_manager.rb:198:in `find_command'
	/Users/loranger/.rubies/ruby-3.1.0/lib/ruby/site_ruby/3.1.0/rubygems/command_manager.rb:183:in `process_args'
	/Users/loranger/.rubies/ruby-3.1.0/lib/ruby/site_ruby/3.1.0/rubygems/command_manager.rb:149:in `run'
	/Users/loranger/.rubies/ruby-3.1.0/lib/ruby/site_ruby/3.1.0/rubygems/gem_runner.rb:51:in `run'
	/Users/loranger/.rubies/ruby-3.1.0/bin/gem:13:in `<main>'
rubygems failed:
   0: Command failed: `/Users/loranger/.rubies/ruby-3.1.0/bin/gem 'update --system' --user-install`
   1: `/Users/loranger/.rubies/ruby-3.1.0/bin/gem` failed: exit status: 1

Location:
   src/steps/generic.rs:100

  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
                                ⋮ 2 frames hidden ⋮
   3: <E as eyre::context::ext::StdError>::ext_report::h3914eb7fca7d3c7e
      at <unknown source file>:<unknown line>
   4: <std::process::Command as topgrade::command::CommandExt>::status_checked_with::h5f958d61cee44d75
      at <unknown source file>:<unknown line>
   5: topgrade::runner::Runner::execute::hc5a7673ee7c1f2ee
      at <unknown source file>:<unknown line>
   6: topgrade::run::h3c67bc3f7c3790f8
      at <unknown source file>:<unknown line>
   7: topgrade::main::hc6c76f70bcfed8aa
      at <unknown source file>:<unknown line>
   8: std::sys_common::backtrace::__rust_begin_short_backtrace::he9796edb044ece37
      at <unknown source file>:<unknown line>
   9: _main<unknown>
      at <unknown source file>:<unknown line>

Run with COLORBT_SHOW_HIDDEN=1 environment variable to disable frame filtering.
Run with RUST_BACKTRACE=full to include source snippets.
Retry? (y)es/(N)o/(s)hell/(q)uit
```

</details>

<details>
<summary>gem update --system --user-install</summary>

```
$ gem update --system --user-install
Latest version already installed. Done.
```

</details>

**Comment by DottoDev** [comment]
Thanks for reporting. I'll look into that. While waiting you could look at the steps that solved the problem for others. See #246

**Comment by bingalls** [comment]
I get an error of
```bash
rubygems failed:
   0: Command failed: `/usr/local/Cellar/ruby/3.1.2_1/bin/gem 'update --system' --user-install`
```
When I run
```bash
/usr/local/Cellar/ruby/3.1.2_1/bin/gem update --system --user-install
```
alone on the command line, it works. Note that I had to remove the quotes around 'update --system'
HTH

**Comment by mouse07410** [comment]
I have the same problem with Macports-installed Ruby.

Hope #248 is merged soonest.

**Comment by DottoDev** [comment]
Closed by #248
